### ISSUE TITLE
Project API performance comparison 2020.6 -> 2021.1

### ISSUE THREAD

**Issue by tothegills** [OctopusDeploy/Issues#6948]
# The bug
Requesting projects in parallel can result in many open transactions and cause some requests to fail.

## What I expected to happen

The same query run against version 2020.6 returned all results in ~20 seconds.

## Steps to reproduce
The following was run against a database with 150 projects on version 2020.6 and version 2021.1

```
const string url = "https://octopus";
var client = await OctopusAsyncClient.Create(new OctopusServerEndpoint(url), new OctopusClientOptions()).ConfigureAwait(false);
await client.SignIn(new LoginCommand { Username = "admin", Password = "password", RememberMe = true }).ConfigureAwait(false);
var repository = client.Repository;
var watch = new Stopwatch();
watch.Start();
var projects = await repository.Projects.GetAll();
projects.Count.Dump();
var tasks = projects.SelectMany(p =>
{
	return new Task[] {
		repository.Projects.Get(p.Id),
		repository.Projects.GetAllChannels(p),
		repository.DeploymentProcesses.Get(p.DeploymentProcessId)
	};
});
await Task.WhenAll(tasks);
watch.Elapsed.Dump();
watch.Stop();
```

### Log excerpt
2021.1 log showing transaction count climbing and requests taking a long time to complete:
```
2021-07-08 03:16:32.9895  19480     73  INFO  Request took 276126ms: GET https://ttpoint1.tentaclearmy.com:8085/latest/api/Spaces-1/projects/Projects-1321 8000000a-0002-c400-b63f-84710c7967bb by Admin
2021-07-08 03:16:32.9895  19480     85  INFO  422 transactions active
2021-07-08 03:16:32.9895  19480     84  INFO  423 transactions active
2021-07-08 03:16:32.9895  19480     80  INFO  424 transactions active
2021-07-08 03:16:33.0143  19480    120  INFO  425 transactions active
2021-07-08 03:16:33.0151  19480     76  INFO  426 transactions active
```

## Affected versions

**Octopus Server:** 2021.1
Fixed in version 2021.1.7595 

## Links
Internal discussion: https://octopusdeploy.slack.com/archives/C026R6Z2DC0/p1625715613447400
Internal: https://trello.com/c/d2eAdK3M/38-project-api-perf-regression

**Comment by gupta-kartik** [comment]
Release Note: Fixed an issue where multiple transactions being concurrently open lead to requests failing and timing out

**Comment by Octobob** [comment]
:tada: The fix for this issue has been released in:

| Release stream | Release     |
| -------------- | ----------- |
| 2021.1 | 2021.1.7595 |
| 2021.2 | 2021.2.6562 |
| 2021.3 | 2021.3.519 |

**Comment by Octobob** [comment]
:tada: The fix for this issue has been released in:

| Release stream | Release     |
| -------------- | ----------- |
| 2021.1 | 2021.1.7595 |
| 2021.2 | 2021.2.6562 |
| 2021.3 | 2021.3.519 |
| 2022.1+ | _all releases_ |
